presentation/src/main/java/com/going/presentation/todo/change/TodoChangeViewModel.kt

```diff
@@ -0,0 +1,118 @@
+package com.going.presentation.todo.change
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.going.domain.entity.request.TodoChangeRequestModel
+import com.going.domain.entity.response.TodoAllocatorModel
+import com.going.domain.entity.response.TodoDetailModel
+import com.going.domain.repository.TodoRepository
+import com.going.presentation.designsystem.edittext.EditTextState
+import com.going.ui.state.UiState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class TodoChangeViewModel @Inject constructor(
+    private val todoRepository: TodoRepository,
+) : ViewModel() {
+
+    private val _todoDetailState = MutableStateFlow<UiState<TodoDetailModel>>(UiState.Empty)
+    val todoDetailState: StateFlow<UiState<TodoDetailModel>> = _todoDetailState
+
+    private val _todoPatchState = MutableSharedFlow<Boolean>()
+    val todoPatchState: SharedFlow<Boolean> = _todoPatchState
+
+    val todo = MutableLiveData("")
+    val memo = MutableLiveData("")
+    val endDate = MutableLiveData("")
+
+    private lateinit var oldTodoInfo: TodoDetailModel
+    private var isSecret: Boolean = false
+
+    private val isTodoAvailable = MutableLiveData(false)
+    private val isMemoAvailable = MutableLiveData(true)
+    val isFinishAvailable = MutableLiveData(false)
+
+    var allocatorModelList: List<TodoAllocatorModel> = listOf()
+
+    var tripId: Long = 0
+    var todoId: Long = 0
+
+    fun setNameState(name: String, state: EditTextState) {
+        todo.value = name
+        isTodoAvailable.value = state == EditTextState.SUCCESS
+        checkIsFinishAvailable()
+    }
+
+    fun setMemoState(memoText: String, state: EditTextState) {
+        memo.value = memoText
+        isMemoAvailable.value = state != EditTextState.OVER
+        checkIsFinishAvailable()
+    }
+
+    fun checkIsFinishAvailable() {
+        isFinishAvailable.value =
+            isTodoAvailable.value == true && isMemoAvailable.value == true && !endDate.value.isNullOrEmpty() && checkIsTodoChanged()
+    }
+
+    private fun checkIsTodoChanged(): Boolean =
+        todo.value != oldTodoInfo.title || endDate.value != oldTodoInfo.endDate || memo.value != oldTodoInfo.memo || checkIsListChanged()
```

Comment:
오,,,되게 코틀린스러운 함수네요!!

Reply:
조금 사소하긴 한데 저는 두가지를 비교한 값을 하나의 값으로 사용하는 경우에는 괄호로 묶어주는 편입니다! 그게 더 가독성이 좋아보이더라구용

```suggestion
    private fun checkIsTodoChanged(): Boolean =
        (todo.value != oldTodoInfo.title) || (endDate.value != oldTodoInfo.endDate) || (memo.value != oldTodoInfo.memo) || checkIsListChanged()
```